### PR TITLE
Feat: Limited Registrations and Validation Link

### PR DESCRIPTION
**Pull request by anshulg954**
→ Limit Number of Registrations : Restrict the number of users.
→ Ask users for a validation link before registration.
→ Unit Test and Integration test improvised on Client Side.

### CODE REVIEW

File: tabpfn_client/prompt_agent.py [hunk 1]
```diff
@@ -46,7 +46,9 @@ def prompt_and_set_token(cls, user_auth_handler: "UserAuthenticationClient"):
             password = getpass.getpass(cls.indent(password_req_prompt))
             password_confirm = getpass.getpass(cls.indent("Please confirm your password: "))
 
-            user_auth_handler.set_token_by_registration(email, password, password_confirm)
+            validation_link = input(cls.indent("Please enter your validation link: "))
```
Comment (SamuelGabriel) [inline tabpfn_client/prompt_agent.py:49]: isn't it actually a `secret`?

**Review by SamuelGabriel — APPROVED** [review]
This looks good if the corresponding PR in tabpfn-server is merged.

Just maybe fix the wording on the `validation link` in the prompt.

File: tabpfn_client/prompt_agent.py (hunk 1 above)
Comment (anshulg954) [reply tabpfn_client/prompt_agent.py:49]: Shall we keep it as a secret_key / secret_code or something!

**Review by liam-sbhoo — APPROVED** [review]

File: tabpfn_client/prompt_agent.py (hunk 1 above)
Comment (anshulg954) [reply tabpfn_client/prompt_agent.py]: Updated to secret code for prompt

**Pull request merged**